src/Cake.Common/Solution/Project/Properties/AssemblyInfoCreator.cs

```diff
@@ -109,9 +110,13 @@ private static AssemblyInfoRegistration GetRegistrations(AssemblyInfoSettings se
             registration.AddString("AssemblyCopyright", "System.Reflection", settings.Copyright);
             registration.AddString("AssemblyTrademark", "System.Reflection", settings.Trademark);
             registration.AddString("Guid", "System.Runtime.InteropServices", settings.Guid);
+            foreach (var assemblyName in settings.InternalsVisibleTo ?? Enumerable.Empty<string>())
```

Comment:
I think it would be a good idea to not treat the InternalsVisibleTo as a registration.

Maybe something like this in the `Create` method:

```csharp
using (var stream = _fileSystem.GetFile(absoluteOutputPath).OpenWrite())
using (var writer = new StreamWriter(stream, System.Text.Encoding.UTF8))
{
    // Write header.
    writer.WriteLine("//------------------------------------------------------------------------------");
    writer.WriteLine("// <auto-generated>");
    writer.WriteLine("//     This code was generated by Cake.");
    writer.WriteLine("// </auto-generated>");
    writer.WriteLine("//------------------------------------------------------------------------------");

    if (registration.Attributes.Count > 0)
    {
        // Write namespaces.
        var namespaces = registration.Namespaces.Select(n => string.Concat("using ", n, ";"));
        if(settings.InternalsVisibleTo.Count > 0)
        {
            // Add namespace for InternalsVisibleTo to the set.
            registration.Namespaces.Add("System.Runtime.CompilerServices");
        }
        foreach (var @namespace in namespaces)
        {
            writer.WriteLine(@namespace);
        }

        writer.WriteLine();

        // Write attributes.
        foreach (var attribute in registration.Attributes)
        {
            writer.WriteLine(string.Concat("[assembly: ", attribute.Key, "(", attribute.Value, ")]"));
        }

        // Write the internals visible to attributes.
        if(settings.InternalsVisibleTo.Count > 0)
        {
            writer.WriteLine();

            // Write attributes.
            foreach (var attribute in settings.InternalsVisibleTo)
            {
                writer.WriteLine(string.Concat("[assembly: InternalsVisibleTo(", attribute, ")]"));
            }
        }        
    }
}
```